packages/core/src/indexing/database-index-api.ts

```diff
@@ -81,6 +98,33 @@ export abstract class DatabaseIndexApi<DateType = Date | number> {
     this.syncApi = api
   }
 
+  /**
+   * Add an index on model fields to the database
+   * @param model
+   * @param indices
+   */
+  async addFieldsIndex(model: StreamID, indices: Array<FieldsIndex>): Promise<void> {
+    await this.dbConnection(INDEXED_MODEL_CONFIG_TABLE_NAME)
+      .update({ indices: JSON.stringify(indices) })
+      .where('model', model.toString())
+  }
+
+  /**
+   * Get a model's fields index
+   * @param model
+   */
+  async getFieldsIndex(model: StreamID): Promise<Array<FieldsIndex> | undefined> {
+    const res = await this.dbConnection(INDEXED_MODEL_CONFIG_TABLE_NAME)
+      .first('indices')
+      .where('model', model.toString())
+
+    if (res) {
+      return JSON.parse(res.indices)
+    } else {
+      undefined
```

Comment:
should this just be an empty array?

Reply:
Yep, originally was doing this to fallthrough, but request indices should override.